spec-tests/src/test/java/org/candlepin/spec/pools/PoolSpecTest.java

```diff
@@ -0,0 +1,625 @@
+/**
+ * Copyright (c) 2009 - 2022 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.spec.pools;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.candlepin.spec.bootstrap.assertions.StatusCodeAssertions.assertForbidden;
+import static org.candlepin.spec.bootstrap.assertions.StatusCodeAssertions.assertNotFound;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.candlepin.ApiException;
+import org.candlepin.dto.api.v1.AttributeDTO;
+import org.candlepin.dto.api.v1.BrandingDTO;
+import org.candlepin.dto.api.v1.ConsumerDTO;
+import org.candlepin.dto.api.v1.ConsumerTypeDTO;
+import org.candlepin.dto.api.v1.EntitlementDTO;
+import org.candlepin.dto.api.v1.OwnerDTO;
+import org.candlepin.dto.api.v1.PoolDTO;
+import org.candlepin.dto.api.v1.ProductDTO;
+import org.candlepin.dto.api.v1.UserDTO;
+import org.candlepin.spec.bootstrap.client.ApiClient;
+import org.candlepin.spec.bootstrap.client.ApiClients;
+import org.candlepin.spec.bootstrap.client.SpecTest;
+import org.candlepin.spec.bootstrap.data.builder.Branding;
+import org.candlepin.spec.bootstrap.data.builder.Owners;
+import org.candlepin.spec.bootstrap.data.builder.Pools;
+import org.candlepin.spec.bootstrap.data.builder.Products;
+import org.candlepin.spec.bootstrap.data.util.StringUtil;
+import org.candlepin.spec.bootstrap.data.util.UserUtil;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.time.OffsetDateTime;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+
+
+@SpecTest
+class PoolSpecTest {
+
+    private OwnerDTO createOwner(ApiClient client) throws ApiException {
+        return client.owners().createOwner(Owners.random());
```

Comment:
I know I am opening the can of worms again, but if we feel the need to be using such utility methods, and all (or at least most) of the private methods in this class seem generic enough to be able to be used in other spec tests, shouldn't we extract them in appropriate utility class(es)? Could be grouped into classes such as `OwnerUtils`, `PoolUtils`, etc.?

Reply:
And the answer might be no here. Maybe it's fine for each test class to define it's own methods for readability

Reply:
I would prefer not in these cases. I begrudgingly accept factory methods for building DTOs that can then be further manipulated or whatever as needed, but I don't know how valuable it is to make a global method that creates such a DTO and does some kind of operation on the CP side of things transparently.

It starts getting into the realm of one thing our unit tests do incorrectly with its global utilities in that we create something, and then we have to modify it, so instead of just doing create -> persist, we do create -> persist -> update -> persist. The unit test the does same amount of work (config + persist), but we've doubled up on the number of remote hits.